source/javascripts/services/request-service.ts

```diff
@@ -0,0 +1,158 @@
+import { getAppSlug } from "./app-service";
+import StringService from "./string-service";
+import { Logger } from "./logger";
+import { StatusType } from "@datadog/browser-logs";
+
+enum RequestServiceMode {
+	Website = "website",
+	Cli = "cli"
+}
+
+interface RequestServiceConfigureOptions {
+	mode?: string;
+	appSlug?: string;
+	logger?: Logger;
+}
+
+class RequestService {
+	public mode = "";
+	public appSlug = "";
+	private logger: Logger | undefined;
+
+	constructor(logger: Logger) {
+		this.configure({ logger });
+	}
+
+	public configure({ mode, appSlug, logger }: RequestServiceConfigureOptions = {}): void {
+		this.mode = mode || (this.modeFromEnvVars() as string);
+		this.appSlug = appSlug || (getAppSlug() as string);
+		this.logger = logger;
+	}
+
+	public async getAppConfigYML(
```

Comment:
I think this method is way too long.

- I think we should create some kind of abstract wrapping for `fetch`. In my opinion this function should look something like
```
try {
let result = await abstractApiCall(url, params, etc.)
} catch (ex) {
// handle call-specific errors
}
// return or do some mapping
```

- Also we should "standardize" general errors (aborted, not found, 500 etc. - maybe we should create custom error classes - a thin layer over Error - so we can later check for them?) and only handle individual failure cases in the different API call methods.